### ISSUE TITLE
Trailing Slash in buildUrl Method Affecting Expo Project for React Native

### ISSUE THREAD

**Issue by brandon-stookey** [openai/openai-node#490]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

Description:
I have identified a potential bug in the `core.ts` `buildUrl` method which seems to specifically affect users working with the Expo project for React Native. This bug results in the generation of a URL with a trailing slash, leading to a 404 error when making requests.

### To Reproduce

**Steps To reproduce:**
- Use the current implementation of the buildUrl method in an Expo project for React Native.
- Observe the generation of the URL with a trailing slash.
- Make a request to the API using this URL.
- Encounter a 404 error due to the trailing slash.

**I was attempting to use the code below from the readme:**
```javascript
import OpenAI from 'openai';

const openai = new OpenAI({
  apiKey: 'My API Key', // defaults to process.env["OPENAI_API_KEY"]
});

async function main() {
  const chatCompletion = await openai.chat.completions.create({
    messages: [{ role: 'user', content: 'Say this is a test' }],
    model: 'gpt-3.5-turbo',
  });
}

main();
```

The above code outputs a request with a URL with a trailing slash `https://api.openai.com/v1/chat/completions/`

**Suggested Fix:**
Based on the [URL constructor documentation](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL), I recommend updating the code in core.ts as follows:
```javascript
// Simplified version for demo purposes
const url = baseUrl(path, this.baseURL); // 'https://api.openai.com/v1/chat/completions'
```

**Additional Context:**
This bug and its fix have been observed in my Expo project for React Native. However, it's important to note that I was unable to reproduce this bug in Google Chrome Dev Tools.

**Temporary workaround:**
```javascript
const openai = new OpenAI({
  apiKey: 'My API Key', // defaults to process.env["OPENAI_API_KEY"],
  fetch: async (url, init) => {
    try {
      const modifiedUrl = url.replace(/\/$/, "");
      const response = await fetch(modifiedUrl, init);

      return response;
    } catch (error) {
      console.log('openai fetch error', error);
    }
  }
});
```

### OS

mac os

### Node version

Node v16.20.0

### Library version

openai v4.17.4

**Comment by rattrayalex** [comment]
React Native is not supported, as documented in the README. More details [here](https://github.com/openai/openai-node/issues/199#issuecomment-1642859127).

See also:

- https://github.com/openai/openai-node/issues/355
- https://github.com/openai/openai-node/issues/348
- https://github.com/openai/openai-node/issues/270

**Comment by brandon-stookey** [comment]
> React Native is not supported, as documented in the README. More details [here](https://github.com/openai/openai-node/issues/199#issuecomment-1642859127).
> 
> See also:
> 
> * [ReactNative: Error: 404 Invalid URL (POST /v1/chat/completions/) #355](https://github.com/openai/openai-node/issues/355)
> * [openai.chat.completions.create throws a 404 Invalid URL error #348](https://github.com/openai/openai-node/issues/348)
> * [Im not sure why but I keep getting this error. #270](https://github.com/openai/openai-node/issues/270)

Thank you for taking the time to respond and for linking the additional resources :)

**Comment by hadnet** [comment]
@rattrayalex but why not?

**Comment by rattrayalex** [comment]
Last we checked, React Native had no way to support file uploads, because it was missing the necessary primitives (and/or they were broken). In one of the previous threads, perhaps linked above, it seemed that they were possibly developing a fix which could have enabled this to work with a shim, but nobody has yet verified that it works.

**Comment by depombo** [comment]
React Native file uploads do not work indeed. However, I was able to get them to work using the [expo file system](https://docs.expo.dev/versions/latest/sdk/filesystem/). In case it is useful to anyone else, I ended up building an openai react native client that supports file upload and chat completion streaming without polyfills by using [react-native-sse](https://github.com/binaryminds/react-native-sse#readme) and the expo file system. The library uses the official node sdk everywhere else along with the same types and API

https://github.com/backmesh/openai-react-native
